src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java

```diff
@@ -34,8 +41,8 @@
 
     @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
     public DiscountResponseVO findById(Long id) {
-        Discount discount = discountRepository.findById(id)
-                .orElseThrow(() -> new EntityNotFoundException(Discount.class, "id", id));
+        discountValidator.checkIfDiscountIsPresentById(id);
```

Comment:
don't need to put this method into validator

use the same approach as everywhere else

dicountRepository.findById(id)
.orElseThrow(...)

Reply:
returned to previous version